### DISCUSSION TITLE
Got the power control working but non of the fields

### DISCUSSION THREAD

**Discussion by LeighAS** [syssi/esphome-soyosource-gtn-virtual-meter#150] [Q&A] [answered]
![image](https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/assets/19148189/d076ed41-1e0c-42d6-a271-4a1bc3fa57d4)
I used the esp32-example.yaml and the homeassistant API instead of MQTT. the screengrab above is currently how my gtn100 appears to home assistant. The GTN type i have is without a screen on the front (it has a qr code in its place) and i also run the oem wifi dongle on stock firmware using their app to confirm power is being set correctly by the spoofed rs485 attached esp32 limiter.

**My question is do I have the wrong setup YAML file for my GTN hardware. I can set the power limit but i can not read any data coming from the soyosource other than the watts that have been requested?**

pls ignore the power into battery from grid (that is coming from a tuya switch that has a meanwell psu charging the lifepo4 battery via a fechterv4 current limiter to enable cccv charging.

**Comment by syssi** [comment, marked as answer]
Your inverter model is one of the **silent** devices which can receive instructions (power demands) but is unable to respond with a `status` to provide interesting measurements like `battery current`, `battery power`, `battery voltage`.

You should use this YAML / solution instead: https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/blob/main/esp8266-display-wifi-version-limiter-example.yaml
This setup uses two different `uart` connections to talk to the limiter port (as before) **and** to talk to the WiFi dongle port to retrieve the measurements / inverter status.

https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/tree/main/docs/wifi-version

**Reply by LeighAS** [reply]
Thanks. So if i read this correctly my esp32 would connect to the rs485 port via a ftdi board as it already is and also connect to the usb via an opto isolator (if thats the chip in your photo) at the sane time?

**Reply by syssi** [reply]
Your description isn't 100% accurate. There is no FTDI involved but yes: There will be two connections to the inverter. One connection via RS485 and a second with a UART-TTL (3.3v) logic level. If you power your ESP using the wifi/display port the isolator isn't required. If you power the ESP using a power supply the opto isolator is recommended!

**Reply by LeighAS** [reply]
okay thanks Syssi, ordered 3x ADUM1201, will wait on their arrival before proceeding further. thanks for all your help.

**Reply by LeighAS** [reply]
so i havent got around to changing this over yet as it was working in the sense that i could request power from a home assistant automation and thats all i needed. Strangely though tonight the ability to request a set power amount has stopped. also the app has been really reluctant to load on my phone which is also strange. the one time i did get it to load i managed to switch the mode to constant battery power to check whether i could request power directly which it could so im happy the power stage isnt broken. but now im very confused. probably time to see whether i can setup the esp32 as above and rule out having to rely on the shonky software that soyosource provide.

**Reply by LeighAS** [reply]
sorry i was being a retard i had emergency power off switched to on which obviously was doing was it said it should but being on my phone screen was hard to read the whole name. its kind of an annoying switch to see so i might hide it.
